app/com/linkedin/drelephant/spark/heuristics/ConfigurationParametersHeuristic.scala

```diff
@@ -0,0 +1,611 @@
+     * @param numCores current number of cores
+     * @param recommendedCores recommended number of cores
+     * @return the recommended memory in bytes.
+     */
+   private def calculateRecommendedMemory(
+       allocatedMemory: Long,
+       jvmUsedMemory: Long,
+       numCores: Int,
+       recommendedCores: Int): Long = {
+     val calculatedMem = (jvmUsedMemory * Math.ceil(recommendedCores / numCores.toDouble) *
+       (1 + DEFAULT_MEMORY_BUFFER_PERCENTAGE) + JvmUsedMemoryHeuristic.reservedMemory).toLong
+     Math.min(Math.max(DEFAULT_MIN_MEMORY, calculatedMem), allocatedMemory)
+   }
+
+   /** If the severity is sigificant (not NONE or LOW). */
+   private def hasSignificantSeverity(severity: Severity): Boolean = {
+     severity != Severity.NONE && severity != Severity.LOW
+   }
+
+   /**
+     * Calculate the total amount of unified memory allocated across all tasks for
+     * the stage.
+     *
+     * @param executorMemory executor memory in bytes
+     * @param memoryFraction spark.memory.fraction
+     * @param executorCores number of executor cores
+     * @param numTasks number of tasks/partitions
+     * @return
+     */
+   private def calculateTotalUnifiedMemory(
+       executorMemory: Long,
+       memoryFraction: Double,
+       executorCores: Int,
+       numTasks: Int): Long = {
+     // amount of unified memory available for each task
+     val unifiedMemPerTask = (executorMemory - SPARK_RESERVED_MEMORY) * memoryFraction /
+       executorCores
+     (unifiedMemPerTask * numTasks).toLong
+   }
+
+   private val REGEX_MATCHER = raw"(\d+)((?:[T|G|M|K])?B?)?"r
+
+   /**
+     * Given a memory value in bytes, convert it to a string with the unit that round to a >0 integer part.
+     *
+     * @param size The memory value in long bytes
+     * @return The formatted string, null if
+     */
+   private def bytesToString(size: Long): String = {
+     val (value, unit) = {
+       if (size >= 2L * GB_TO_BYTES) {
+         (size.asInstanceOf[Double] / GB_TO_BYTES, "GB")
+       } else {
+         (size.asInstanceOf[Double] / MB_TO_BYTES, "MB")
+       }
+     }
+     s"${Math.ceil(value).toInt}${unit}"
+   }
+
+   /**
+     * Convert a formatted string into a long value in bytes. If no units
+     * are specified, then the default is MB.
+     *
+     * @param formattedString The string to convert
+     * @return The bytes value
+     */
+   private def stringToBytes(formattedString: String): Long = {
+     if (formattedString == null || formattedString.isEmpty) {
+       return 0L
+     }
+     //handling if the string has , for eg. 1,000MB
+     val regularizedString = formattedString.replace(",", "").toUpperCase
+     regularizedString match {
+       case REGEX_MATCHER(value, unit) =>
+         val num = value.toLong
+         if (unit.length == 0) {
+           num * MB_TO_BYTES
+         } else {
+           unit.charAt(0) match {
+             case 'T' => num * TB_TO_BYTES
```

Comment:
You can use FileUtils.ONE_GB, ONE_MB etc.

Reply:
Replaced.